File: LittleProjects/Scala/AkkaCRUD/src/main/scala/service/Service.scala
package service

import model.{Flux, Mono}

trait Service[T] {
  def select(id: Long): Mono[T]

  def selectAll(): Flux[T]

  def insert(entity: T): Mono[T]

  def update(entity: T): Mono[T]

  def delete(id: Long): Mono[T]
}

File: LittleProjects/Java/SpringHibernateCRUD/src/main/java/core/service/Service.java
package core.service;

import java.io.Serializable;

public interface Service<T, ID extends Serializable> {
    T select(ID id);

    Iterable<T> select();

    void deleteById(ID id);

    void deleteById(Iterable<ID> ids);

    void delete(T entity);

    void delete(Iterable<T> entities);

    T insert(T entity);

    Iterable<T> insert(Iterable<T> entities);

    T update(T entity);

    Iterable<T> update(Iterable<T> entities);
}
